changelog/WOOPMNT-4879

```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: dev
+
+Stop using dynamic keys for the database cache in favor of better validation and transients.
```

Comment:
The entry log mentions transients but `Database_Cache` still uses `update_option`

Reply:
For once I decided to add the changelog entry in the initial commits of the PR, and this happened! 🤣

I'll update it.